packages/astro/src/cli/index.ts

```diff
@@ -133,8 +133,8 @@ async function runCommand(cmd: string, flags: yargs.Arguments) {
 		}
 		case 'sync': {
 			const { sync } = await import('./sync/index.js');
-			const exitCode = await sync({ flags });
-			return process.exit(exitCode);
+			await sync({ flags });
+			return;
```

Comment:
We already have a global handler for CLI that handles more stuff (telemetry, error formatting etc) so this wasn't really useful.